framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -910,6 +911,13 @@ public AnnotatedTypeMirror getAnnotatedType(Tree tree) {
     protected void postProcessClassTree(ClassTree tree) {
         TypesIntoElements.store(processingEnv, this, tree);
         DeclarationsIntoElements.store(processingEnv, this, tree);
+        if (checker.getOptions().containsKey("inferSignatures")) {
```

Comment:
As mentioned above, I would instantiate SignatureInferenceScenes in the ATF, write results here, and reclaim any unused memory.